control/cli.py

```diff
@@ -1409,65 +1410,36 @@ def ns_resize(self, args):
         return ret.status
 
     def format_size(self, sz):
-        units = ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB']
-        unit_index = 0
-        while sz >= 1024 and unit_index < len(units) - 1:
+        units = ["Bytes"] + GatewayClient.SIZE_UNITS
+        for unit_index in range(len(units)):
+            if sz < 1024:
+                break
             sz /= 1024.0
-            unit_index += 1
+        unit = f"{units[unit_index]}iB" if unit_index > 0 else f"{units[unit_index]}"
         if sz == int(sz):
-            return f"{int(sz)} {units[unit_index]}"
-        return f"{sz:2.1f} {units[unit_index]}"
+            return f"{int(sz)} {unit}"
+        return f"{sz:2.1f} {unit}"
 
     def get_size_in_bytes(self, sz):
         multiply = 1
         sz = sz.strip()
-        if sz.endswith("K"):
-            sz = sz[:-1]
-            multiply = 1024
-        elif sz.endswith("KB"):
-            sz = sz[:-2]
-            multiply = 1000
-        elif sz.endswith("KiB"):
-            sz = sz[:-3]
-            multiply = 1024
-        elif sz.endswith("M"):
-            sz = sz[:-1]
-            multiply = 1024 * 1024
-        elif sz.endswith("MB"):
-            sz = sz[:-2]
-            multiply = 1000 * 1000
-        elif sz.endswith("MiB"):
-            sz = sz[:-3]
-            multiply = 1024 * 1024
-        elif sz.endswith("G"):
-            sz = sz[:-1]
-            multiply = 1024 * 1024 * 1024
-        elif sz.endswith("GB"):
-            sz = sz[:-2]
-            multiply = 1000 * 1000 * 1000
-        elif sz.endswith("GiB"):
-            sz = sz[:-3]
-            multiply = 1024 * 1024 * 1024
-        elif sz.endswith("T"):
-            sz = sz[:-1]
-            multiply = 1024 * 1024 * 1024 * 1024
-        elif sz.endswith("TB"):
-            sz = sz[:-2]
-            multiply = 1000 * 1000 * 1000 * 1000
-        elif sz.endswith("TiB"):
-            sz = sz[:-3]
-            multiply = 1024 * 1024 * 1024 * 1024
-        elif sz.endswith("P"):
-            sz = sz[:-1]
-            multiply = 1024 * 1024 * 1024 * 1024 * 1024
-        elif sz.endswith("PB"):
-            sz = sz[:-2]
-            multiply = 1000 * 1000 * 1000 * 1000 * 1000
-        elif sz.endswith("PiB"):
-            sz = sz[:-3]
-            multiply = 1024 * 1024 * 1024 * 1024 * 1024
+        for unit_index in range(len(GatewayClient.SIZE_UNITS)):
+            found = False
+            if sz.endswith(GatewayClient.SIZE_UNITS[unit_index]):
+                sz = sz[:-1]
+                found = True
+            elif sz.endswith(GatewayClient.SIZE_UNITS[unit_index] + "iB"):
+                sz = sz[:-3]
+                found = True
+            elif sz.endswith(GatewayClient.SIZE_UNITS[unit_index] + "B"):
+                sz = sz[:-2]
+                found = True
```

Comment:
@gbregman treating MiB and MB as if it is the same number of bytes is not a good idea 😊

Reply:
@baum this is how Aviv wanted it. The fact that we had both TB and TiB before just created confusion with QA. Nobody there understood the difference.